### PR TITLE
Migrate some dependency

### LINKED ISSUE #28: ModuleNotFoundError: No module named 'imp'
#### Description

Installation fails on Python 3.12

#### Steps/Code to Reproduce

`pip install pyrubberband`

#### Expected Results

Works

#### Actual Results

```python
Collecting pyrubberband
  Downloading pyrubberband-0.3.0.tar.gz (4.1 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [20 lines of output]
      Traceback (most recent call last):
        File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/local/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/mw/d2jp6b715v59p7ghfyp6rr5c0000gn/T/pip-build-env-ry4g175i/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 325, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=['wheel'])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/mw/d2jp6b715v59p7ghfyp6rr5c0000gn/T/pip-build-env-ry4g175i/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 295, in _get_build_requires
          self.run_setup()
        File "/private/var/folders/mw/d2jp6b715v59p7ghfyp6rr5c0000gn/T/pip-build-env-ry4g175i/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 487, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/private/var/folders/mw/d2jp6b715v59p7ghfyp6rr5c0000gn/T/pip-build-env-ry4g175i/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 311, in run_setup
          exec(code, locals())
        File "<string>", line 3, in <module>
      ModuleNotFoundError: No module named 'imp'
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error
```

#### Versions

```
macOS-14.4.1-x86_64-i386-64bit
Python 3.12.3 (main, Apr 12 2024, 20:23:48) [Clang 15.0.0 (clang-1500.1.0.2.5)]
NumPy 1.26.4
```

### PR DESCRIPTION
**Pull request by jhj0517**
#### Reference Issue
Fix #28

#### What does this implement/fix? Explain your changes.
1. Migrate to `importlib` from `imp`. 
Done by : https://github.com/python/cpython/issues/104212
2. `pysoundfile`  is now deprecated and use `soundfile`. ( It was problematic if other packages used `soundfile`.)

Tested on python `3.10`, `3.11`, `3.12` and all worked fine.

### CODE REVIEW

File: .travis_dependencies.sh
Comment (bmcfee) [inline .travis_dependencies.sh:37]: This is fine, but really we need to migrate this project from travis to gh actions.  As is, I don't think the tests work anymore.

File: setup.py
Comment (bmcfee) [inline setup.py]: This whole thing can be simplified quite a bit by migrating to setup.cfg.  Resampy has a reasonable example of how this should look, and we can bypass all the imp/importlib business:
https://github.com/bmcfee/resampy/blob/main/setup.cfg

**Review by bmcfee — APPROVED** [review]
TLDR: thanks for doing this!  It all seems fine as is, but it would be nice to modernize the package a bit beyond just the soundfile dependency so that CI continues working.

File: setup.py
Comment (jhj0517) [reply setup.py]: Thanks for the reference! Migration to [setup.cfg](https://github.com/bmcfee/pyrubberband/pull/29/files#diff-fa602a8a75dc9dcc92261bac5f533c2a85e34fcceaff63b3a3a81d9acde2fc52) is done. 
Tested on `python 3.6` ~ `python 3.12` and confirmed that all works fine.

File: .travis_dependencies.sh
Comment (jhj0517) [reply .travis_dependencies.sh:37]: I added a [simple action](https://github.com/bmcfee/pyrubberband/pull/29/files#diff-34152c90777df3b17754e08caa876570ec218ca4782df47eb66ee10b6cf48e52R1) that installs dependencies and runs `python -m pytest tests` on `python 3.6` ~ `python 3.12`.

You can see the [result of the build](https://github.com/jhj0517/pyrubberband/actions/runs/11032672136) from the test branch.

+) This [jhj0517:fix/compatibility](https://github.com/jhj0517/pyrubberband/tree/fix/compatibility) branch is allowed to be directly edited by maintainers, so you may be able to remove old `.travis` files if you want.

File: pyrubberband/__init__.py
Comment (jhj0517) [inline pyrubberband/__init__.py:7]: I had to do this because of the `python 3.6` problem.

Using `version = attr: pyrubberband.version.version` from the setup.cfg tried to read actual script and import packages in the `__init__.py` before installing packages and it caused `ModuleNotFoundError`.

Putting the version on top solved the problem. Refer to : [stackoverflow disscussion](https://stackoverflow.com/a/71495114/27460890)

This only occurs in `python 3.6`.

**Review by bmcfee — APPROVED** [review]

**Comment by bmcfee** [comment]
LGTM, thanks!  I'll try to finish the CI cleanup after merging this.

**Pull request merged**
